docs/design/phases.md

```diff
@@ -0,0 +1,127 @@
+
+### Generate KubeConfig files on the master
+
+There should be:
+ - a KubeConfig file for kubeadm to use itself and the admin: `/etc/kubernetes/admin.conf`
+   - the "admin" here is defined as `kubeadm` itself and the actual person(s) that is administering the cluster and want to control the cluster
+     - with this file, the admin has full control over the cluster 
+   - inside this file, a client certificate is generated from the `ca.crt` and `ca.key`. The client cert should:
+     - be a client certificate (`x509.ExtKeyUsageClientAuth`)
+     - be in the `system:masters` organization
+ - a KubeConfig file for kubelet to use: `/etc/kubernetes/kubelet.conf`
+   - inside this file, a client certificate is generated from the `ca.crt` and `ca.key`. The client cert should:
+     - be a client certificate (`x509.ExtKeyUsageClientAuth`)
+     - be in the `system:nodes` organization
+
+`ca.crt` is also embedded in the KubeConfig files.
+
+### Generate a token.csv file for bootstrapping purposes
+
+There should be:
+ - a `/etc/kubernetes/tokens.csv` file that contains one kubeadm-generated token
+   - the content should be in the form `${token},${user},${uuid},${group}` where:
+     - `${token}` is the full token `123456.0123456789abcdef`
+     - `${user}` is currently `kubeadm-node-csr`, should be changed
+     - `${uuid}` is a random UUID created with `"k8s.io/apimachinery/pkg/util/uuid".NewUUID()`
+     - `${group}` is currently `kubeadm:kubelet-bootstrap`, might be changed
+
+### Bootstrap the control plane
+
+First, determine if there is an etcd cluster available:
+ - use an external one if external etcd params are given
+ - spin up one locally if external etcd is not be enabled. It should:
+   - be spun up in a Pod
+   - listen on `localhost:2379`
+   - set `PodSpec.SecurityContext.SELinuxOptions.Type=spc_t`
+   - be at a minimum version `3.0.14`
+   - make a volume out to the host on a given host path
+
+Then, spin up the control plane as static pods:
+ - They are located in `/etc/kubernetes/manifests`
+ - The files are stored as `.yaml` files.
+ - `HostNetwork: true` is present on all Pods since there is no network configured yet
+
+Then kubeadm waits for the control plane to come up as static pods
+
+The master node is patched, the master label is added, and the master is tainted.
+
+The proposed master label for `v1.6` is `node-role.kubernetes.io/master=true`.
+Nodes may also register themselves with the label `node-role.kubernetes.io/node=true` by using the `--node-labels` flag.
+
+The taint for proposed beta use is:
+
+```yaml
+apiVersion: v1
+kind: Node
+spec:
+  taints:
+  - key: node-role.kubernetes.io
+    value: master
+    effect: NoSchedule
+```
+
+If self-hosting should be enabled, do a little bit more:
+ - TODO
+
+### Configure the bootstrap tokens
+
+This phase configures the BootstrapSigner to create the public `cluster-info` ConfigMap in the `kube-public` namespace
```

Comment:
Can you please provide a link, even if to just the PR, to what the BootstrapSigner is?